include/aikido/planner/vectorfield/VectorFieldUtil.hpp

```diff
@@ -41,47 +34,83 @@ class DesiredTwistFunction : public dart::optimizer::Function
-  /// \param[out] _grad Gradient of a defined objective
+  /// \param[in] qd Joint velocities.
+  /// \param[out] grad Gradient of a defined objective
   void evalGradient(
-      const Eigen::VectorXd& _qd, Eigen::Map<Eigen::VectorXd> _grad) override;
+      const Eigen::VectorXd& qd, Eigen::Map<Eigen::VectorXd> grad) override;
 
 private:
+  /// Twist.
   Twist mTwist;
+
+  /// Jacobian of Meta Skeleton
   Jacobian mJacobian;
 };
 
 /// Compute joint velocity from a given twist.
 ///
-/// \param[in] _desiredTwist Desired twist, which consists of angular velocity
-/// and linear velocity
-/// \param[in] _stateSpace MetaSkeleton state space
-/// \param[in]  _bodyNode Body node of the end-effector
-/// \param[in] _optimizationTolerance Callback of vector field calculation
-/// \param[in] _timestep How long will the computed joint velocities be executed
-/// \param[in] _padding Padding for joint limits
-/// \param[out] _jointVelocity Calculated joint velocities
+/// \param[out] jointVelocity Calculated joint velocities.
+/// \param[in] desiredTwist Desired twist, which consists of angular velocity
+/// and linear velocity.
+/// \param[in] stateSpace MetaSkeleton state space.
+/// \param[in] bodyNode Body node of the end-effector.
+/// \param[in] jointLimitPadding If less then this distance to joint
+/// limit, velocity is bounded in that direction to 0.
+/// \param[in] jointVelocityLowerLimits Joint velocity lower bounds.
+/// \param[in] jointVelocityUpperLimits Joint velocity upper bounds.
+/// \param[in] jointVelocityLimited Whether joint velocity bounds are
+/// considered in optimization.
+/// \param[in] stepSize Step size in second.
+/// \return Whether a joint velocity is found
 bool computeJointVelocityFromTwist(
-    const Eigen::Vector6d& _desiredTwist,
-    const aikido::statespace::dart::MetaSkeletonStateSpacePtr _stateSpace,
-    const dart::dynamics::BodyNodePtr _bodyNode,
-    double _optimizationTolerance,
-    double _timestep,
-    double _padding,
-    Eigen::VectorXd& _jointVelocity);
+    Eigen::VectorXd& jointVelocity,
+    const Eigen::Vector6d& desiredTwist,
+    const aikido::statespace::dart::MetaSkeletonStateSpacePtr stateSpace,
+    const dart::dynamics::BodyNodePtr bodyNode,
+    double jointLimitPadding,
+    const Eigen::VectorXd& jointVelocityLowerLimits,
+    const Eigen::VectorXd& jointVelocityUpperLimits,
+    bool jointVelocityLimited,
+    double stepSize);
+
+/// Compute the twist in global coordinate that corresponds to the gradient of
+/// the geodesic distance between two transforms.
+///
+/// \param[in] currentTrans Current transformation.
+/// \param[in] goalTrans Goal transformation.
+Eigen::Vector6d computeGeodesicTwist(
+    const Eigen::Isometry3d& currentTrans, const Eigen::Isometry3d& goalTrans);
+
+/// Compute the error in gloabl coordinate between two transforms.
+///
+/// \param[in] currentTrans Current transformation.
+/// \param[in] goalTrans Goal transformation.
```

Comment:
Add a docstring for the return value. Particularly: document what order its elements are in.